### ISSUE TITLE
Train/Infer With Various Image Sizes

### ISSUE THREAD

**Issue by ertugrul-dmr** [ultralytics/yolov5#8498]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have training images coming from different sources with different image sizes, but mainly 512x512. But there are other images in training which 768x512, 640x512 etc.

I can train them in squared format or rect format and it should be ok. But my problem is in the inference part. I need to deliver my results in pascal_voc format (x_min, y_min, x_max, y_max) and test images comes in different shapes. My models metrics will be controlled in this format (IoU over original image sized boxes), which I can de-normalize them back from yolo format easily. 

My main concern is the yolov5 giving these normalized detection bbox coordinates before or after the resizing done in the inference part? Let's say I infer at 512x512, then I de-normalize coordinates using the original image shapes (768x512, 512x512, etc), would results be valid for the original bounding boxes in test set?

Thanks in advance...

**Comment by glenn-jocher** [comment]
@ertugrul-dmr 👋 Hello! Thanks for asking about **handling inference results**. Your use case is standard, you can use default training and inference. `results.xyxy` are returned in original image coordinates.

YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!

**Comment by ertugrul-dmr** [comment]
Thank you for your answer!

One last thing, if I want to infer in batches do I need to resize them?

**Comment by glenn-jocher** [comment]
@ertugrul-dmr no, resizing in batches is handled automatically
